php-frontend/src/main/java/org/sonar/php/parser/NewPHPGrammar.java

```diff
@@ -0,0 +1,1011 @@
+
+  private final GrammarBuilder<InternalSyntaxToken> b;
+  private final TreeFactory f;
+
+  public NewPHPGrammar(GrammarBuilder<InternalSyntaxToken> b, TreeFactory f) {
+    this.b = b;
+    this.f = f;
+  }
+
+  /**
+   * [ START ] Declaration
+   */
+
+  public VariableDeclarationTree MEMBER_CONST_DECLARATION() {
+    return b.<VariableDeclarationTree>nonterminal(PHPLexicalGrammar.MEMBER_CONST_DECLARATION)
+        .is(f.memberConstDeclaration(
+            b.token(PHPLexicalGrammar.IDENTIFIER),
+            b.optional(f.newTuple18(b.token(EQU), STATIC_SCALAR()))
+        ));
+  }
+
+  public NamespaceNameTree NAMESPACE_NAME() {
+    return b.<NamespaceNameTree>nonterminal(PHPLexicalGrammar.NAMESPACE_NAME)
+        .is(b.firstOf(
+          f.namespaceName(
+            b.optional(b.token(PHPPunctuator.NS_SEPARATOR)),
+            b.zeroOrMore(f.newTuple4(
+              b.firstOf(b.token(PHPLexicalGrammar.IDENTIFIER)),
+              b.token(PHPPunctuator.NS_SEPARATOR)
+            )),
+            b.token(PHPLexicalGrammar.IDENTIFIER)),
+          f.namespaceName(
+              b.token(PHPKeyword.NAMESPACE),
+              b.token(PHPPunctuator.NS_SEPARATOR),
+              b.zeroOrMore(f.newTuple6(
+                  b.firstOf(b.token(PHPLexicalGrammar.IDENTIFIER)),
+                  b.token(PHPPunctuator.NS_SEPARATOR)
+              )),
+              b.token(PHPLexicalGrammar.IDENTIFIER)))
+    );
+  }
+  
+  public UseDeclarationTree USE_DECLARATION() {
+    return b.<UseDeclarationTree>nonterminal(PHPLexicalGrammar.USE_DECLARATION).is(
+      f.useDeclaration(
+        NAMESPACE_NAME(),
+        b.optional(
+          f.newTuple91(
+            b.token(PHPKeyword.AS), 
+            b.token(PHPLexicalGrammar.IDENTIFIER)))));
+  }
+  
+  public InternalSyntaxToken MEMBER_MODIFIER() {
+    return b.<InternalSyntaxToken>nonterminal().is(
+      b.firstOf(
+        b.token(PHPKeyword.PUBLIC),
+        b.token(PHPKeyword.PROTECTED),
+        b.token(PHPKeyword.PRIVATE),
+        b.token(PHPKeyword.STATIC),
+        b.token(PHPKeyword.ABSTRACT),
+        b.token(PHPKeyword.FINAL)
+      ));
+  }
+
+  public MethodDeclarationTree METHOD_DECLARATION() {
+    return b.<MethodDeclarationTree>nonterminal(PHPLexicalGrammar.METHOD_DECLARATION).is(
+      f.methodDeclaration(
+        b.<SyntaxToken>zeroOrMore(MEMBER_MODIFIER()),
+        b.token(PHPKeyword.FUNCTION),
+        b.optional(b.token(PHPPunctuator.AMPERSAND)),
+        IDENTIFIER(),
+        PARAMETER_LIST(),
+        b.firstOf(
+          EOS(),
+          BLOCK())
+      ));
+  }
+
+  public FunctionDeclarationTree FUNCTION_DECLARATION() {
+    return b.<FunctionDeclarationTree>nonterminal(PHPLexicalGrammar.FUNCTION_DECLARATION).is(
+      f.functionDeclaration(
+        b.token(PHPKeyword.FUNCTION),
+        b.optional(b.token(PHPPunctuator.AMPERSAND)),
+        IDENTIFIER(),
+        PARAMETER_LIST(),
+        BLOCK()));
+  }
+  
+  public ParameterListTree PARAMETER_LIST() {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS00100)